src/hooks/useFavoriteState.ts

```diff
@@ -0,0 +1,59 @@
+'use client';
+
+import { useCallback, useMemo } from 'react';
+import useCookie from 'react-use-cookie';
+import { useRouter } from 'next/navigation';
+
+export function useFavoriteState(initialFavorites: string[]) {
+  const router = useRouter();
+  const defaultValue = initialFavorites ? JSON.stringify(initialFavorites) : '[]';
+  const [cookieValue, setCookieValue] = useCookie('relisten_favorites:artists', defaultValue);
```

Comment:
Had a really hard time DRYing the `relisten_favorites:artists` variable to reuse in `src/lib/serverFavoriteCookies.ts`. It kept saying the export was a function?

Also open to changing naming convention. Figure it was worth scoping to artist so a different cookie name can be used for [tracks](https://github.com/RelistenNet/relisten-web/issues/88).